src/Init/Data/Format/Basic.lean

```diff
@@ -300,11 +300,16 @@ instance : MonadPrettyFormat (StateM State) where
   startTag _         := return ()
   endTags _          := return ()
 
-/-- Pretty-print a `Format` object as a string with expected width `w`. -/
+/--
+Renders a `Format` to a string.
+* `w`: the total width
+* `indent`: the initial indentation
+* `column`: the initial column for the first line
```

Comment:
So far my experiments suggest that `indent` does nothing, and column is actually the column spacing.

In any case, in the stuff I want to upstream this is being called with `indent := 0` and `column := 0`, so I'm just going to ditch this!

Reply:
FWIW, @PatrickMassot and I recently found a need for `indent` and annoyingly could not use it because `Format.State` is private (could we also un`private` it?).

`indent` is useful when we are starting to pretty-print in the middle of a larger piece of syntax, for instance to programmatically edit a tactic proof.

`indent` is the amount of indentation that should be prepended to each new line produced by `pretty`. For instance if we already have
```lean
. someTactic
  $fmt
```
and wish to substitute `pretty fmt` for `$fmt`, where `fmt : Format`, we ought to pretty-print with `indent := 2`. If `indent := 0` and `fmt` is `rw`, we might end up with
```lean
. someTactic
  rw [
lemma ]
```

`column` does have an effect, for instance in `#eval pretty (text "_" ++ (nest 4 $ align false ++ text "=>")) (indent := 0) (column := 0)`, but I cannot make sense of what its semantics are supposed to be.

Reply:
Okay, got in sorted out (thanks @thorimur for a quick conversation on this), and hopefully the new doc-string is much clearer.

Reply:
@Vtec234 You can use `Std.Format.prettyExtra` for that (it is used in TryThis). It is implemented using `open private Format.State` though.